feature/gribi/otg_tests/vrf_policy_driven_te/vrf_policy_driven_te_test.go

```diff
@@ -0,0 +1,1946 @@
+	iDut6Ipv4 := iDut6Eth.Ipv4Addresses().Add().SetName(atePort6.Name + ".IPv4")
+	iDut6Ipv4.SetAddress(atePort6.IPv4).SetGateway(dutPort6.IPv4).SetPrefix(uint32(atePort6.IPv4Len))
+	iDut6Ipv6 := iDut6Eth.Ipv6Addresses().Add().SetName(atePort6.Name + ".IPv6")
+	iDut6Ipv6.SetAddress(atePort6.IPv6).SetGateway(dutPort6.IPv6).SetPrefix(uint32(atePort6.IPv6Len))
+
+	iDut7Dev := config.Devices().Add().SetName(atePort7.Name)
+	iDut7Eth := iDut7Dev.Ethernets().Add().SetName(atePort7.Name + ".Eth").SetMac(atePort7.MAC)
+	iDut7Eth.Connection().SetChoice(gosnappi.EthernetConnectionChoice.PORT_NAME).SetPortName(port7.Name())
+	iDut7Ipv4 := iDut7Eth.Ipv4Addresses().Add().SetName(atePort7.Name + ".IPv4")
+	iDut7Ipv4.SetAddress(atePort7.IPv4).SetGateway(dutPort7.IPv4).SetPrefix(uint32(atePort7.IPv4Len))
+	iDut7Ipv6 := iDut7Eth.Ipv6Addresses().Add().SetName(atePort7.Name + ".IPv6")
+	iDut7Ipv6.SetAddress(atePort7.IPv6).SetGateway(dutPort7.IPv6).SetPrefix(uint32(atePort7.IPv6Len))
+
+	iDut8Dev := config.Devices().Add().SetName(atePort8.Name)
+	iDut8Eth := iDut8Dev.Ethernets().Add().SetName(atePort8.Name + ".Eth").SetMac(atePort8.MAC)
+	iDut8Eth.Connection().SetChoice(gosnappi.EthernetConnectionChoice.PORT_NAME).SetPortName(port8.Name())
+	iDut8Ipv4 := iDut8Eth.Ipv4Addresses().Add().SetName(atePort8.Name + ".IPv4")
+	iDut8Ipv4.SetAddress(atePort8.IPv4).SetGateway(dutPort8.IPv4).SetPrefix(uint32(atePort8.IPv4Len))
+	iDut8Ipv6 := iDut8Eth.Ipv6Addresses().Add().SetName(atePort8.Name + ".IPv6")
+	iDut8Ipv6.SetAddress(atePort8.IPv6).SetGateway(dutPort8.IPv6).SetPrefix(uint32(atePort8.IPv6Len))
+	// Configure Loopback on port8.
+	iDut8LoopV4 := iDut8Dev.Ipv4Loopbacks().Add().SetName("Port8LoopV4").SetEthName(iDut8Eth.Name())
+	iDut8LoopV4.SetAddress(otgIsisPort8LoopV4)
+	iDut8LoopV6 := iDut8Dev.Ipv6Loopbacks().Add().SetName("Port8LoopV6").SetEthName(iDut8Eth.Name())
+	iDut8LoopV6.SetAddress(otgIsisPort8LoopV6)
+
+	// Enable ISIS and BGP Protocols on port 8.
+	isisDut := iDut8Dev.Isis().SetName("ISIS1").SetSystemId(otgSysID1)
+	isisDut.Basic().SetIpv4TeRouterId(atePort8.IPv4).SetHostname(isisDut.Name()).SetLearnedLspFilter(true)
+	isisDut.Interfaces().Add().SetEthName(iDut8Dev.Ethernets().Items()[0].Name()).
+		SetName("devIsisInt1").
+		SetLevelType(gosnappi.IsisInterfaceLevelType.LEVEL_2).
+		SetNetworkType(gosnappi.IsisInterfaceNetworkType.POINT_TO_POINT)
+
+	// Advertise OTG Port8 loopback address via ISIS.
+	isisPort2V4 := iDut8Dev.Isis().V4Routes().Add().SetName("ISISPort8V4").SetLinkMetric(10)
+	isisPort2V4.Addresses().Add().SetAddress(otgIsisPort8LoopV4).SetPrefix(32)
+	isisPort2V6 := iDut8Dev.Isis().V6Routes().Add().SetName("ISISPort8V6").SetLinkMetric(10)
+	isisPort2V6.Addresses().Add().SetAddress(otgIsisPort8LoopV6).SetPrefix(uint32(128))
+
+	iDutBgp := iDut8Dev.Bgp().SetRouterId(otgIsisPort8LoopV4)
+	iDutBgp4Peer := iDutBgp.Ipv4Interfaces().Add().SetIpv4Name(iDut8LoopV4.Name()).Peers().Add().SetName(atePort8.Name + ".BGP4.peer")
```

Comment:
is iBGP over loopback interface required ? same scenario can work with iBGP over connected interfaces

Reply:
It is done as per README

Reply:
in the readme it just says BGP route resolved by ISIS in default VRF, so iBGP session can be over connected interface as well.